spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -191,6 +192,20 @@ public Blob createBlob() throws StorageException {
 		return this.storage.create(BlobInfo.newBuilder(getBlobId()).build());
 	}
 
+	/**
+	 * Creates the blob that this {@link GoogleStorageResource} represents in Google Cloud
+	 * Storage and fills it with provided content.
+	 * @param contents the initial file contents to write
+	 * @return the created blob object
+	 * @throws StorageException if any errors during blob creation arise,
+	 * such as if the blob already exists
+	 * @throws IllegalStateException if the resource reference is to a bucket, and not a blob.
+	 * @since 1.3
```

Comment:
Doesn't seem right if it's in the 1.2 branch.
Remove this method?

Reply:
Hmm. It's useful and does not introduce any breaking changes.
Perhaps I should remove `@since` in the master instead?

Reply:
AFAIK, only bug-fixes should be introduced in service releases.

Reply:
@artembilan Happy New Year! Are Spring Cloud's service releases expected to contain only bugfixes, or are backwards-compatible new features okay?

This PR is backporting a 1.3.x change that really contains two changes addressing the same issue:
1) A bugfix: making resource writing a one-step operation when an output stream is acquired, as the previous behavior of creating the file and then acquiring the stream failed unless the user had a delete permission (because replacement is considered a removal and subsequent addition in GCP storage).
2) A new convenience feature allowing the object to be created with content in a single call, without having to acquire an output stream first.

So, we definitely need to backport (1), but are not sure about (2).

Reply:
We are OK with new feature in the services release as long as they don't break anything and existing application can upgrade without noticing new stuff.

Do not back port feature is just a choice with less pain during maintenance: such a new feature has to be fixed now in several versions 😄 (if there are going to be bugs of course...)